aries_cloudagent/anoncreds/issuer.py

```diff
@@ -505,11 +508,8 @@ async def cred_def_supports_revocation(self, cred_def_id: str) -> bool:
 
     async def create_and_register_revocation_registry_definition(
         self,
-        issuer_id: str,
-        cred_def_id: str,
-        registry_type: str,
-        tag: str,
-        max_cred_num: int,
+        profile,
+        issuer_rev_reg_record: IssuerRevRegRecord,
```

Comment:
Could you help me understand the motivation for passing the issuer rev reg record instead of the individual components of issuer_id, cred_def_id, etc.?

Reply:
My motivation was that if we pass in the whole record, we can access the components that aren't directly passed in (`_id`, `state`) that previously we didn't have to pass in because the method was on the `IssuerRevRegRecord` itself. This seemed preferrable to adding `_id` and `state` to be retrieved from the request body and passed directly into `AnonCredsIssuer.create_and_register_revocation_registry_definition()`. Curious to hear your thoughts and other alternatives.